cscs-checks/microbenchmarks/cpu/simd/nsimd.py

```diff
@@ -7,98 +7,52 @@
+        std_mul/f32                       491 ns          491 ns      1425407
+        MIPP_mul/f32                     83.5 ns         83.5 ns      8386285
 
-    > reframe --system dom:gpu -p PrgEnv-gnu -r -c nsimd.py
-        PERFORMANCE REPORT
-        -----------------------------------------------------------------------
-        NsimdTestt
-        - dom:gpu
-           - PrgEnv-gnu
-            * speedup: 4.254 x (ns)
-
-    > Test code is:
-        # std_sqrt (sqrt.avx2.f64.cpp:308)
-        *(_r + i) = std::sqrt(*(_0 + i));
-
-        # nsimd_cpu_sqrt (sqrt.avx2.f64.cpp:104)
-        return _mm256_sqrt_pd(a0);
-        nsimd::storea(_r + i,
-                      nsimd::sqrt(nsimd::loada(_0 + i, f64(), nsimd::cpu()),
-                                  f64(),
-                                  nsimd::cpu()),
-                      f64(),
-                      nsimd::cpu()  );
-
-        # nsimd_avx2_sqrt (sqrt.avx2.f64.cpp:138)
-        _mm256_sqrt_pd (avxintrin.h:1039)
-
-        # nsimd_avx2_sqrt_unroll2
-        nsimd::storea(_r + i, nsimd::sqrt(nsimd::loada<nsimd::pack<f64, 2>
-                      >(_0 + i)));
-
-        # nsimd_avx2_sqrt_unroll3
-        nsimd::storea(_r + i, nsimd::sqrt(nsimd::loada<nsimd::pack<f64, 3>
-                      >(_0 + i)));
-
-        # nsimd_avx2_sqrt_unroll4
-        nsimd::storea(_r + i, nsimd::sqrt(nsimd::loada<nsimd::pack<f64, 4>
-                      >(_0 + i)));
-
-        # Sleef_Sleef_sqrtd4_avx2
-        nsimd::storea(_r + i, Sleef_sqrtd4_avx2(nsimd::loada(_0 + i, f64())),
-                      f64());
-
-        # MIPP_sqrt
-        mipp::store(_r + i, mipp::sqrt<f64>(mipp::load<f64>(_0 + i)));
+    Example PERFORMANCE REPORT
+    --------------------------
+    NsimdTest_mul_avx2_f32_cpp
+    - eiger:mc
+       - builtin
+          * num_tasks: 1
+          * speedup: 12.306 x (ns)
     '''
-    def __init__(self):
-        self.valid_systems = ['dom:mc', 'dom:gpu']
+    def __init__(self, testname):
+        self.valid_systems = ['dom:mc', 'dom:gpu', 'eiger:mc']
         self.valid_prog_environs = ['builtin']
-        self.descr = 'sqrt.avx2.f64 example'
+        self.descr = f'testing {testname}'
         self.build_system = 'SingleSource'
-        self.testname = 'sqrt.avx2.f64'
-        self.sourcesdir = None
-        self.srcdir = 'benches/cxx_adv'
-        self.sourcepath = '%s/%s.cpp' % (self.srcdir, self.testname)
-        self.prebuild_cmds = [
-            'tar xf $EBROOTNSIMD/benches.tar benches/benches.hpp',
-            'tar xf $EBROOTNSIMD/benches.tar %s' % self.sourcepath,
-        ]
-        self.executable = '%s.exe' % self.testname
-        self.modules = ['nsimd/579084-CrayGNU-19.10']
+        self.testname = testname
+        # c++ test code generated with:
+        #   python3 egg/hatch.py --benches --simd avx2 -m mul
+        self.sourcesdir = 'benches'
+        self.sourcepath = self.testname
```

Comment:
Just to be consistent with the line below.

```suggestion
        self.sourcepath = testname
```